src/foam/mlang/mlang.js

```diff
@@ -807,6 +840,18 @@ foam.CLASS({
       }
 
       return rhs ? rhs.indexOf(lhs) !== -1 : false;
+    },
+    function toDisjunctiveNormalForm() {
+      foam.assert(this.Constant.isInstance(this.arg2),
```

Comment:
A non-constant arg2 could still produce a viable array to use for IN, so this is going to limit its functionality. Breaking IN down into EQ's would be fine for cases where it is possible, but the fallback would be to leave the IN where it is (for indexes, for instance, IN can be optimized)